src/blob.rs

```diff
@@ -538,7 +538,7 @@ impl<'a> Iterator for BlobDirIter<'a> {
     type Item = BlobObject<'a>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        while let Some(path) = self.iter.next() {
+        for path in self.iter.as_ref() {
```

Comment:
are you sure this doesn't restart every time?

Reply:
yeah.... `s/as_ref/by_ref/`  TIL